UI/Dialogs/BaseDialog.cpp

```diff
@@ -1086,4 +1116,85 @@ namespace dialog
 	{
 		return m_lpMapiObjects;
 	}
+	
+	std::wstring LookupFamilyName(_In_ PCWSTR familyName, _In_ const UINT32 filter)
+	{
+		UINT32 count = 0;
+		UINT32 length = 0;
+		std::wstring buildInfo;
+		LONG rc = FindPackagesByPackageFamily(familyName, filter, &count, nullptr, &length, nullptr, nullptr);
+		if (rc == ERROR_SUCCESS)
+		{
+			output::DebugPrint(1, L"FindPackagesByPackageFamily -> No packages found\n");
+			return L"";
+		}
+		else if (rc != ERROR_INSUFFICIENT_BUFFER)
+		{
+			output::DebugPrint(1, L"Error %d in FindPackagesByPackageFamily\n");
+			return L"";
+		}
+
+		PWSTR * fullNames = (PWSTR *)malloc(count * sizeof(*fullNames));
+		if (fullNames == nullptr)
+		{
+			free(fullNames);
+			output::DebugPrint(1, L"LookupFamilyName -> Error allocating memory\n");
+			return L"";
+		}
+
+		PWSTR buffer = (PWSTR)malloc(length * sizeof(WCHAR));
+		if (buffer == nullptr)
+		{
+			free(buffer);
+			free(fullNames);
+			output::DebugPrint(1, L"LookupFamilyName -> Error allocating memory\n");
+			return L"";
+		}
+
+		UINT32 * properties = (UINT32 *)malloc(count * sizeof(*properties));
+		if (properties == nullptr)
+		{
+			free(properties);
+			free(buffer);
+			free(fullNames);
+			output::DebugPrint(1, L"LookupFamilyName -> Error allocating memory\n");
+			return L"";
+		}
+
+		rc = FindPackagesByPackageFamily(familyName, filter, &count, fullNames, &length, buffer, properties);
+		if (rc != ERROR_SUCCESS)
+		{
+			output::DebugPrint(1, L"FindPackagesByPackageFamily -> Error %d looking up Full Names from Family Names\n");
+			free(properties);
+			free(buffer);
+			free(fullNames);
+			return L"";
+		}
+		else
+		{
+			for (UINT32 index = 0; index < count; ++index)
+			{
+				buildInfo = L"%u: %s [0x%X %s]\n", index, fullNames[index], properties[index], PackagePropertiesToString(properties[index]);
```

Comment:
Does this rebuild buildInfo on each pass of the loop? Did you intend to concatenate them? Also - I've not seen this notation for formatting a string - should there be a string::format wrapping this?

Reply:
This is what was used in the article:

https://msdn.microsoft.com/en-us/library/windows/desktop/dn270601(v=vs.85).aspx

Reply:
In the article, these were parameters to printf.  Here, they’re just comma separated tokens. Try using strings::format

Reply:
thanks for the help, is this more in line with what you were looking for?

buildInfo = strings::format(L"%u: %s [0x%X %s]\n", index, fullNames[index], properties[index], PackagePropertiesToString(properties[index]));